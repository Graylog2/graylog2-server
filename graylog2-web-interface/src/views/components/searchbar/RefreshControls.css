:local(.position) {
    margin-right: 15px;
}