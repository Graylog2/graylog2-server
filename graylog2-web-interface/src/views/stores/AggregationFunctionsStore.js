import Reflux from 'reflux';

import URLUtils from 'util/URLUtils';
import fetch from 'logic/rest/FetchProvider';
import { singletonStore } from 'views/logic/singleton';

const functionsUrl = URLUtils.qualifyUrl('/views/functions');

export default singletonStore(
  'views.AggregationFunctions',
  () => Reflux.createStore({
    init() {
      this.refresh();
    },

    getInitialState() {
      return this._state();
    },

    refresh() {
      fetch('GET', functionsUrl).then((response) => {
        this.functions = response;
        this._trigger();
      });
    },
    _state() {
      return this.functions;
    },
    _trigger() {
      this.trigger(this._state());
    },
  }),
);
