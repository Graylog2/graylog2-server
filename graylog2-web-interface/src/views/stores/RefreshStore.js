// @flow strict
import Reflux from 'reflux';

import SearchActions from 'views/actions/SearchActions';
import { singletonActions, singletonStore } from 'views/logic/singleton';

type RefreshActionsType = {
  enable: () => void,
  disable: () => void,
  setInterval: (number) => void,
};

export const RefreshActions: RefreshActionsType = singletonActions(
  'views.Refresh',
  () => Reflux.createActions([
    'enable',
    'disable',
    'setInterval',
  ]),
);

export const RefreshStore = singletonStore(
  'views.Refresh',
  () => Reflux.createStore({
    listenables: [RefreshActions],

    refreshConfig: {},

    intervalId: undefined,

    init() {
      this.refreshConfig = {
        enabled: false,
        interval: 1000,
      };
    },

    getInitialState() {
      return this.refreshConfig;
    },

    setInterval(interval: number) {
      this.refreshConfig = { interval, enabled: true };

      if (this.intervalId) {
        clearInterval(this.intervalId);
        this.intervalId = undefined;
      }

      this.intervalId = setInterval(SearchActions.executeWithCurrentState, this.refreshConfig.interval);
      this._trigger();
    },

    enable() {
      this.refreshConfig = { ...this.refreshConfig, enabled: true };

      if (!this.intervalId) {
        this.intervalId = setInterval(SearchActions.executeWithCurrentState, this.refreshConfig.interval);
      }

      this._trigger();
    },

    disable() {
      this.refreshConfig = { ...this.refreshConfig, enabled: false };

      if (this.intervalId) {
        clearInterval(this.intervalId);
        this.intervalId = undefined;
      }

      this._trigger();
    },

    _trigger() {
      const { enabled, interval } = this.refreshConfig;

      this.trigger({ enabled, interval });
    },
  }),
);
