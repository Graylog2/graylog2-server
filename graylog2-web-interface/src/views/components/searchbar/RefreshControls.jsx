// @flow strict
import React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
import styled from 'styled-components';

import connect from 'stores/connect';
import { MenuItem, ButtonGroup, DropdownButton, Button } from 'components/graylog';
import { Icon, Pluralize } from 'components/common';
import { RefreshActions, RefreshStore } from 'views/stores/RefreshStore';

const ControlsContainer = styled.div`
  max-width: 100%;
`;

const FlexibleButtonGroup = styled(ButtonGroup)`
  display: flex;

  > .btn-group {
    .btn:first-child {
      max-width: 100%;
    }
  }
`;

const ButtonLabel = styled.div`
  display: inline-block;
  text-overflow: ellipsis;
  overflow: hidden;
  max-width: calc(100% - 9px);
  vertical-align: inherit;
`;

type RefreshConfig = {
  interval: number,
  enabled: boolean,
};

type Props = {
  refreshConfig: RefreshConfig,
};

class RefreshControls extends React.Component<Props> {
  static propTypes = {
    refreshConfig: PropTypes.object.isRequired,
  };

  static INTERVAL_OPTIONS: Array<[string, number]> = [
    ['1 Second', 1000],
    ['2 Seconds', 2000],
    ['5 Seconds', 5000],
    ['10 Seconds', 10000],
    ['30 Seconds', 30000],
    ['1 Minute', 60000],
    ['5 Minutes', 300000],
  ];

  componentWillUnmount(): void {
    RefreshActions.disable();
  }

  _toggleEnable = (): void => {
    const { refreshConfig } = this.props;

    if (refreshConfig.enabled) {
      RefreshActions.disable();
    } else {
      RefreshActions.enable();
    }
  };

  _onChange = (interval: number): void => {
    RefreshActions.setInterval(interval);
  };

  _buttonLabel = (refreshConfigEnabled, naturalInterval) => {
    let buttonText = 'Not updating';

    if (refreshConfigEnabled) {
      buttonText = <>Every {naturalInterval}</>;
    }

    return <ButtonLabel>{buttonText}</ButtonLabel>;
  }

  render() {
    const { refreshConfig } = this.props;
    const intervalOptions = RefreshControls.INTERVAL_OPTIONS.map(([label, interval]: [string, number]) => {
      return <MenuItem key={`RefreshControls-${label}`} onClick={() => this._onChange(interval)}>{label}</MenuItem>;
    });
    const intervalDuration = moment.duration(refreshConfig.interval);
    const naturalInterval = intervalDuration.asSeconds() < 60
      ? <span>{intervalDuration.asSeconds()} <Pluralize singular="second" plural="seconds" value={intervalDuration.asSeconds()} /></span>
      : <span>{intervalDuration.asMinutes()} <Pluralize singular="minute" plural="minutes" value={intervalDuration.asMinutes()} /></span>;
    const buttonLabel = this._buttonLabel(refreshConfig.enabled, naturalInterval);

    return (
      <ControlsContainer className="pull-right">
        <FlexibleButtonGroup>
          <Button onClick={this._toggleEnable}>
            {refreshConfig.enabled ? <Icon name="pause" /> : <Icon name="play" />}
          </Button>

          <DropdownButton title={buttonLabel} id="refresh-options-dropdown">
            {intervalOptions}
          </DropdownButton>
        </FlexibleButtonGroup>
      </ControlsContainer>
    );
  }
}

export default connect(RefreshControls, { refreshConfig: RefreshStore });
