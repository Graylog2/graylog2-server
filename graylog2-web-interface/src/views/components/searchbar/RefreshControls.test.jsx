import React from 'react';
import { mount } from 'wrappedEnzyme';
import 'helpers/mocking/react-dom_mock';

import { RefreshActions } from 'views/stores/RefreshStore';
import RefreshControls from './RefreshControls';

jest.useFakeTimers();

jest.mock('stores/connect', () => Component => props => <Component {...(Object.assign({}, props, {}))} />);
jest.mock('views/stores/RefreshStore', () => ({
  RefreshActions: {
    enable: jest.fn(),
    disable: jest.fn(),
  },
  RefreshStore: {},
}));


describe('RefreshControls', () => {
  describe('rendering', () => {
    const verifyRendering = ({ enabled, interval }) => {
      const wrapper = mount(<RefreshControls refreshConfig={{ enabled, interval }} />);
      expect(wrapper).toMatchSnapshot();
    };

    it.each`
    enabled      | interval
    ${true}      | ${1000}
    ${true}      | ${2000}
    ${true}      | ${5000}
    ${true}      | ${10000}
    ${true}      | ${30000}
    ${true}      | ${60000}
    ${true}      | ${300000}
    ${false}     | ${300000}
    ${false}     | ${1000}
  `('it renders refresh controlls with enabled: $enabled and interval: $interval', verifyRendering);
  });

  it('should start the interval', () => {
    const wrapper = mount(<RefreshControls refreshConfig={{ enabled: false, interval: 1000 }} />);
    wrapper.find('i.fa-play').simulate('click');
    expect(RefreshActions.enable).toHaveBeenCalled();
  });

  it('should stop the interval', () => {
    const wrapper = mount(<RefreshControls refreshConfig={{ enabled: true, interval: 1000 }} />);
    wrapper.find('i.fa-pause').simulate('click');
    expect(RefreshActions.disable).toHaveBeenCalled();
  });
});
