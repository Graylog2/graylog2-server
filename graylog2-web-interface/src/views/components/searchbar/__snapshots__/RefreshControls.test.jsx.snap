// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefreshControls rendering it renders refresh controlls with enabled: false and interval: 1000 1`] = `
<div
  className="pull-right RefreshControls__ControlsContainer-sc-1pygk8n-0 kShVuS"
>
  <div
    className="RefreshControls__FlexibleButtonGroup-sc-1pygk8n-1 etyOYS btn-group"
  >
    <button
      className="Button__StyledButton-c9cbmb-0 kRDLih btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-play"
      />
    </button>
    <div
      className="dropdown btn-group"
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="DropdownButton-sc-1343dcx-0 kgGfWL dropdown-toggle btn btn-default"
        disabled={false}
        id="refresh-options-dropdown"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        <div
          className="RefreshControls__ButtonLabel-sc-1pygk8n-2 jztvlP"
        >
          Not updating
        </div>
         
        <span
          className="caret"
        />
      </button>
      <ul
        aria-labelledby="refresh-options-dropdown"
        className="dropdown-menu"
        role="menu"
      >
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            1 Second
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            2 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            5 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            10 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            30 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            1 Minute
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            5 Minutes
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`RefreshControls rendering it renders refresh controlls with enabled: false and interval: 300000 1`] = `
<div
  className="pull-right RefreshControls__ControlsContainer-sc-1pygk8n-0 kShVuS"
>
  <div
    className="RefreshControls__FlexibleButtonGroup-sc-1pygk8n-1 etyOYS btn-group"
  >
    <button
      className="Button__StyledButton-c9cbmb-0 kRDLih btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-play"
      />
    </button>
    <div
      className="dropdown btn-group"
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="DropdownButton-sc-1343dcx-0 kgGfWL dropdown-toggle btn btn-default"
        disabled={false}
        id="refresh-options-dropdown"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        <div
          className="RefreshControls__ButtonLabel-sc-1pygk8n-2 jztvlP"
        >
          Not updating
        </div>
         
        <span
          className="caret"
        />
      </button>
      <ul
        aria-labelledby="refresh-options-dropdown"
        className="dropdown-menu"
        role="menu"
      >
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            1 Second
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            2 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            5 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            10 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            30 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            1 Minute
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            5 Minutes
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`RefreshControls rendering it renders refresh controlls with enabled: true and interval: 1000 1`] = `
<div
  className="pull-right RefreshControls__ControlsContainer-sc-1pygk8n-0 kShVuS"
>
  <div
    className="RefreshControls__FlexibleButtonGroup-sc-1pygk8n-1 etyOYS btn-group"
  >
    <button
      className="Button__StyledButton-c9cbmb-0 kRDLih btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-pause"
      />
    </button>
    <div
      className="dropdown btn-group"
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="DropdownButton-sc-1343dcx-0 kgGfWL dropdown-toggle btn btn-default"
        disabled={false}
        id="refresh-options-dropdown"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        <div
          className="RefreshControls__ButtonLabel-sc-1pygk8n-2 jztvlP"
        >
          Update every 
          <span>
            1
             
            <span>
              second
            </span>
          </span>
        </div>
         
        <span
          className="caret"
        />
      </button>
      <ul
        aria-labelledby="refresh-options-dropdown"
        className="dropdown-menu"
        role="menu"
      >
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            1 Second
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            2 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            5 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            10 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            30 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            1 Minute
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            5 Minutes
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`RefreshControls rendering it renders refresh controlls with enabled: true and interval: 2000 1`] = `
<div
  className="pull-right RefreshControls__ControlsContainer-sc-1pygk8n-0 kShVuS"
>
  <div
    className="RefreshControls__FlexibleButtonGroup-sc-1pygk8n-1 etyOYS btn-group"
  >
    <button
      className="Button__StyledButton-c9cbmb-0 kRDLih btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-pause"
      />
    </button>
    <div
      className="dropdown btn-group"
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="DropdownButton-sc-1343dcx-0 kgGfWL dropdown-toggle btn btn-default"
        disabled={false}
        id="refresh-options-dropdown"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        <div
          className="RefreshControls__ButtonLabel-sc-1pygk8n-2 jztvlP"
        >
          Update every 
          <span>
            2
             
            <span>
              seconds
            </span>
          </span>
        </div>
         
        <span
          className="caret"
        />
      </button>
      <ul
        aria-labelledby="refresh-options-dropdown"
        className="dropdown-menu"
        role="menu"
      >
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            1 Second
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            2 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            5 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            10 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            30 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            1 Minute
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            5 Minutes
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`RefreshControls rendering it renders refresh controlls with enabled: true and interval: 5000 1`] = `
<div
  className="pull-right RefreshControls__ControlsContainer-sc-1pygk8n-0 kShVuS"
>
  <div
    className="RefreshControls__FlexibleButtonGroup-sc-1pygk8n-1 etyOYS btn-group"
  >
    <button
      className="Button__StyledButton-c9cbmb-0 kRDLih btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-pause"
      />
    </button>
    <div
      className="dropdown btn-group"
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="DropdownButton-sc-1343dcx-0 kgGfWL dropdown-toggle btn btn-default"
        disabled={false}
        id="refresh-options-dropdown"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        <div
          className="RefreshControls__ButtonLabel-sc-1pygk8n-2 jztvlP"
        >
          Update every 
          <span>
            5
             
            <span>
              seconds
            </span>
          </span>
        </div>
         
        <span
          className="caret"
        />
      </button>
      <ul
        aria-labelledby="refresh-options-dropdown"
        className="dropdown-menu"
        role="menu"
      >
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            1 Second
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            2 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            5 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            10 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            30 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            1 Minute
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            5 Minutes
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`RefreshControls rendering it renders refresh controlls with enabled: true and interval: 10000 1`] = `
<div
  className="pull-right RefreshControls__ControlsContainer-sc-1pygk8n-0 kShVuS"
>
  <div
    className="RefreshControls__FlexibleButtonGroup-sc-1pygk8n-1 etyOYS btn-group"
  >
    <button
      className="Button__StyledButton-c9cbmb-0 kRDLih btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-pause"
      />
    </button>
    <div
      className="dropdown btn-group"
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="DropdownButton-sc-1343dcx-0 kgGfWL dropdown-toggle btn btn-default"
        disabled={false}
        id="refresh-options-dropdown"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        <div
          className="RefreshControls__ButtonLabel-sc-1pygk8n-2 jztvlP"
        >
          Update every 
          <span>
            10
             
            <span>
              seconds
            </span>
          </span>
        </div>
         
        <span
          className="caret"
        />
      </button>
      <ul
        aria-labelledby="refresh-options-dropdown"
        className="dropdown-menu"
        role="menu"
      >
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            1 Second
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            2 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            5 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            10 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            30 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            1 Minute
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            5 Minutes
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`RefreshControls rendering it renders refresh controlls with enabled: true and interval: 30000 1`] = `
<div
  className="pull-right RefreshControls__ControlsContainer-sc-1pygk8n-0 kShVuS"
>
  <div
    className="RefreshControls__FlexibleButtonGroup-sc-1pygk8n-1 etyOYS btn-group"
  >
    <button
      className="Button__StyledButton-c9cbmb-0 kRDLih btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-pause"
      />
    </button>
    <div
      className="dropdown btn-group"
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="DropdownButton-sc-1343dcx-0 kgGfWL dropdown-toggle btn btn-default"
        disabled={false}
        id="refresh-options-dropdown"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        <div
          className="RefreshControls__ButtonLabel-sc-1pygk8n-2 jztvlP"
        >
          Update every 
          <span>
            30
             
            <span>
              seconds
            </span>
          </span>
        </div>
         
        <span
          className="caret"
        />
      </button>
      <ul
        aria-labelledby="refresh-options-dropdown"
        className="dropdown-menu"
        role="menu"
      >
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            1 Second
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            2 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            5 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            10 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            30 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            1 Minute
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            5 Minutes
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`RefreshControls rendering it renders refresh controlls with enabled: true and interval: 60000 1`] = `
<div
  className="pull-right RefreshControls__ControlsContainer-sc-1pygk8n-0 kShVuS"
>
  <div
    className="RefreshControls__FlexibleButtonGroup-sc-1pygk8n-1 etyOYS btn-group"
  >
    <button
      className="Button__StyledButton-c9cbmb-0 kRDLih btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-pause"
      />
    </button>
    <div
      className="dropdown btn-group"
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="DropdownButton-sc-1343dcx-0 kgGfWL dropdown-toggle btn btn-default"
        disabled={false}
        id="refresh-options-dropdown"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        <div
          className="RefreshControls__ButtonLabel-sc-1pygk8n-2 jztvlP"
        >
          Update every 
          <span>
            1
             
            <span>
              minute
            </span>
          </span>
        </div>
         
        <span
          className="caret"
        />
      </button>
      <ul
        aria-labelledby="refresh-options-dropdown"
        className="dropdown-menu"
        role="menu"
      >
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            1 Second
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            2 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            5 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            10 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            30 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            1 Minute
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            5 Minutes
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`RefreshControls rendering it renders refresh controlls with enabled: true and interval: 300000 1`] = `
<div
  className="pull-right RefreshControls__ControlsContainer-sc-1pygk8n-0 kShVuS"
>
  <div
    className="RefreshControls__FlexibleButtonGroup-sc-1pygk8n-1 etyOYS btn-group"
  >
    <button
      className="Button__StyledButton-c9cbmb-0 kRDLih btn btn-default"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <i
        className="fa fa-pause"
      />
    </button>
    <div
      className="dropdown btn-group"
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="DropdownButton-sc-1343dcx-0 kgGfWL dropdown-toggle btn btn-default"
        disabled={false}
        id="refresh-options-dropdown"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        type="button"
      >
        <div
          className="RefreshControls__ButtonLabel-sc-1pygk8n-2 jztvlP"
        >
          Update every 
          <span>
            5
             
            <span>
              minutes
            </span>
          </span>
        </div>
         
        <span
          className="caret"
        />
      </button>
      <ul
        aria-labelledby="refresh-options-dropdown"
        className="dropdown-menu"
        role="menu"
      >
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            1 Second
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            2 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            5 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            10 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            30 Seconds
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            1 Minute
          </a>
        </li>
        <li
          className=""
          role="presentation"
        >
          <a
            href="#"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="menuitem"
            tabIndex="-1"
          >
            5 Minutes
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
